python/cuda_cccl/cuda/cccl/_cuda_version_utils.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""
+CUDA version detection utilities shared across the cccl package.
+"""
+
+import os
+import shutil
+import subprocess
+from pathlib import Path
+from typing import Optional
+
+
+def detect_cuda_version() -> Optional[int]:
+    """
+    Detect the CUDA major version from the installed toolkit.
+    Returns the major version (12 or 13) or None if not found.
+    """
+    cuda_path = get_cuda_path()
+    if cuda_path:
+        return get_cuda_version_from_path(cuda_path)
```

Comment:
The whole file looks very much biased toward the old way (local CTK centric). I think this should be as simple as getting the `cuda-bindings` major version. 
- on the wheel side, this is then fully controlled by `[cu12]` or `[cu13]`
- on the conda side, likewise controlled by `cuda-version==12.*` or `13.*`
- the rest can go away

Reply:
No. We only attempt to load driver/nvrtc/etc when they are actually used. So it's importable on CPU-only machines (we test this on conda-forge).

Reply:
We probably should add `cuda-bindings` to the required dependency (but make it unconstrained) to ensure it is installed and importable

Reply:
Anyway, I prefer the simplicity of using `bindings`. If it backfires on us we can revisit.

Reply:
> Doesn't this risk installing cuda-bindings 13 into a CUDA 12 environment?

It is already possible today unfortunately, I doubt this would make it worse but it is fine either way.

Reply:
I went with complaining at runtime if `cuda.bindings` is not installed: https://github.com/NVIDIA/cccl/pull/5613/commits/22745af1f8d4f84290c25cc01c5e98023ab0cef1

It's easier for the user to install something that's not installed, than to uninstall the wrong thing and reinstall it.